File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        nodes = []

        def iot(node):
            if node:
                iot(node.left)
                nodes.append(node.val)
                iot(node.right)

        iot(root)
        return nodes

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    // do a in order traversal to store all the nodes in an array
    // index the k - 1 element
    public int kthSmallest(TreeNode root, int k) {
        ArrayList<Integer> nodes = new ArrayList<Integer>();
        inOrderAddNodesToArray(root, nodes);
        return nodes.get(k - 1);
    }   

    public void inOrderAddNodesToArray(TreeNode root, List nodes) {
        if (root != null) {
            inOrderAddNodesToArray(root.left, nodes);
            nodes.add(root.val);
            inOrderAddNodesToArray(root.right, nodes);
        }
    }
}
